rrule/src/parser/content_line/content_line_parts.rs

```diff
@@ -0,0 +1,117 @@
+use crate::parser::{regex::get_property_name, ParseError};
+
+use super::PropertyName;
+
+#[derive(Debug, PartialEq, Clone)]
+pub(crate) struct ContentLineCaptures {
+    pub property_name: PropertyName,
+    pub parameters: Option<String>,
```

Comment:
I expect `parameters` be a hashmap (or any other kind of iter). Cannot we parse parameters at this stage, in `get_content_line_parts`?

Reply:
I forgot that `value` is not actually a `HashMap`, for example `DTSTART;VALUE=DATE:19970101` where `value` is  `19970101` which is just a string.

I attempted to make the `parameters` a `HashMap<K, &str>`, where K is `DateParameter` (for `RDATE`,`DTSTART` etc ) or `RuleParameter` (for `RRULE`, `EXRULE`). The problem is that the generic K has to be provided by the caller of `get_content_line_parts`, which doesn't know what kind of line it is yet so can't possibly know what K should be!

Another solution would be to use `HashMap<&str, &str>` and then later convert that to `HashMap<DateParameter, &str>`, but I think at that point it is just easier to keep `parameters` as a `&str`.

Reply:
Since the only caller of this is `Grammer`. I think it will be fine if we use `&str` everywhere. At least for this iteration.